src/util/file_log.rs

```diff
@@ -0,0 +1,209 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::time::{SystemTime, Duration, UNIX_EPOCH};
+use time::{self, Timespec, Tm};
+use std::fs::{self, OpenOptions, File};
+use std::io::{self, Write};
+use std::path::Path;
+use std::sync::Mutex;
+use log::{self, LogLevelFilter, Log, LogMetadata, LogRecord, SetLoggerError};
+
+use super::logger;
+
+const ONE_DAY_SECONDS: u64 = 60 * 60 * 24;
+const NANOSECONDS_PER_MILLISECOND: i32 = 1_000_000;
+
+fn systemtime_to_tm(t: SystemTime) -> Tm {
+    let duration = t.duration_since(UNIX_EPOCH).unwrap();
+    let spec = Timespec::new(duration.as_secs() as i64, duration.subsec_nanos() as i32);
+    time::at(spec)
+}
+
+fn compute_rollover_time(tm: Tm) -> Tm {
+    let day_start_tm = Tm {
+        tm_hour: 0,
+        tm_min: 0,
+        tm_sec: 0,
+        tm_nsec: 0,
+        ..tm
+    };
+    let duration = time::Duration::from_std(Duration::new(ONE_DAY_SECONDS, 0)).unwrap();
+    (day_start_tm.to_utc() + duration).to_local()
```

Comment:
`Tm.ADD()` requires `self` to be in UTC, as the result. Without `to_utc()` and `to_local()`, the time zone info would be cast away and it results the wrong value.